docs/tutorial.md

```diff
@@ -0,0 +1,88 @@
+# Tutorial Study: Fake button study.
+
+Learner goals:
+
+1. build a shield-instrumented Embedded Web Extension
+2. understand how probes, analysis, code relate
+
+Concepts:
+
+* technical terms
+  * web extension
+  * embedded web extension
+  * Legacy (bootstrap) add-on
+* debugging techniques
+  * `web-ext`
+  * `about:debugging`
+  * `run-firefox.js`
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Contents**
+
+- [Questions?](#questions)
+- [project goal](#project-goal)
+- [Analysis](#analysis)
+- [Engineering steps](#engineering-steps)
+  - [WebExtension with a button](#webextension-with-a-button)
+  - [Multiple Button Choices](#multiple-button-choices)
+  - [Send Message for Telemetry](#send-message-for-telemetry)
+  - [Embedded WebExtension](#embedded-webextension)
+  - [incorporating shield to listen](#incorporating-shield-to-listen)
+- [Advanced](#advanced)
+  - [Non embedded webExtension](#non-embedded-webextension)
+- [good probes](#good-probes)
+- [Why Shield-Utils?](#why-shield-utils)
+- [surveys?](#surveys)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Questions?
+
+* start from empty dir, or from the finished project?
+* how much to build in the first step.
+* how to display the order of the steps?
+
+(Steps, order unclear)
+
+## project goal
+
+## Analysis
+
+## Engineering steps
+
+### WebExtension with a button
+
+### Multiple Button Choices
+
+Make a class that has a startup at random.
+
+### Send Message for Telemetry
+
+Have a sendMessage language for this.
+
+Work on TELEMETRY.md
+
+### Embedded WebExtension
+
+Plumb the listener
+
+### incorporating shield to listen
+
+* PERMANENT Randomization, revisited.
+* Telemetry sending
+* debugging
+* helper add-on
+
+## Advanced
+
+### Non embedded webExtension
+
+## good probes
+
+* good probes mirror the analysis.
+* Analysis wants to be sql. Try to make the probes reflect that.
+
+## Why Shield-Utils?
+
+## surveys?
```

Comment:
Nit: Surveys.

Since we're only changing case in all these headings, I don't think we need to regen the TOC (unless you change the wording, obviously).